hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -367,6 +363,20 @@ private long acquireLock() throws UnknownHostException, TException, InterruptedE
     return lockId;
   }
 
+  private void cleanupMetadataAndUnlock(boolean errorThrown, String metadataLocation, Optional<Long> lockId) {
+    try {
+      if (errorThrown) {
+        // if anything went wrong, clean up the uncommitted metadata file
+        io().deleteFile(metadataLocation);
+      }
+    } catch (RuntimeException e) {
+      LOG.error("Fail to cleanup metadata file at {}", metadataLocation, e);
+      throw e;
```

Comment:
Not a fan of Throw within a Catch.
Maybe you can wrap it in something else.

Reply:
I think this is fine because we don't want to wrap or modify the original exception, and it is helpful to log the metadata location that was not correctly deleted.

Reply:
I think we may leverage `runSafely` that was added recently.

```
    ExceptionUtil.runSafely(
        () -> {
          if (errorThrown) {
            // if anything went wrong, clean up the uncommitted metadata file
            io().deleteFile(metadataLocation);
          }
          return Void.TYPE;
        },
        e -> LOG.error("Failed to cleanup metadata file at {}", metadataLocation, e),
        () -> unlock(lockId));
```

Reply:
cc @rdblue @jackye1995 @giovannifumarola